keps/sig-cloud-provider/2395-removing-in-tree-cloud-providers/README.md

```diff
@@ -195,153 +242,387 @@ import (
+Kubernetes documentation:
+* [Cloud Controller Manager Administration][ccmadmin]
+* [Migrate Replicated Control Plane To Use Cloud Controller Manager][ccmmigrate]
+
+###### What specific metrics should inform a rollback?
+There are a few observable behaviors that inform a rollback. If users start to
+see nodes failing to register, load balancer calls failing or not happening,
+deleted nodes not being removed, or components failing to start due to missing
+taint tolerations, these could all be possible signs that a CCM rollout has
+failed.
+
+In addition there are several metrics exposed by the cloud-provider framework
+that is maintained by SIG Cloud Provider. If the external CCM is using the
+cloud-provider framework, the following metrics could inform about failing
+behavior, users should be observant of these metric values increasing over
+a short period of time, or after upgrade:
+
+* `cloud_provider_taint_removal_delay_seconds`
+* `initial_node_sync_delay_seconds`
+* `nodesync_error_total`
+* `nodesync_latency_seconds`
+* `update_loadbalancer_host_latency_seconds`
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+Yes, the upgrade and downgrade paths have been tested manually on AWS and GCE
+by those provider owners. We do not have artifacts of those tests.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+* The provider options (e.g. `aws`, `azure`, etc), with the exception of `external`,
+  for the `--cloud-provider` flag of `kubelet`, `kube-apiserver`, and
+  `kube-controller-manager` are deprecated with the beta release.
+* The in-tree cloud controller loops for `kubelet`, `kube-apiserver`, and
+  `kube-controller-manager` are deprecated with the beta release.
+
+### Monitoring Requirements
 
-* **What are other known failure modes?**
+###### How can an operator determine if the feature is in use by workloads?
+* In genreal, if pods are being scheduled to the cluster then this feature is
+  working as intended, and is in use by workloads.
+* Workloads that explicitly use load balancer type services will depend on external CCMs
+  on providers that support the load balancer service controller.
+* Workloads that depend on well-known zone and region labels on nodes are
+  also considered to be using this feature transitively.
+
+###### How can someone using this feature know that it is working for their instance?
+- [x] Other (treat as last resort)
+  - Details: The `Uninitialized` taint should be removed from Node objects.
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+* Nodes being initialized properly, zone and region labels being added.
+* Pods being scheduled.
+* Load balancer type services being created and destroyed.
+* The existing SLOs for functionalities the `kube-controller-manager` should be used,
+  this change is supposed to be no-op from the end-user perspective.
```

Comment:
SLOs are about metrics (and their thresholds) - this is too hand-wavy answer for the question.

Please either be more specific, or just leave the last bullet point...

Reply:
i've replaced the vague statements with something more concrete that references metric values.